src/components/ContentItem/contentItem.tsx

```diff
@@ -6,62 +6,68 @@ import remarkGfm from 'remark-gfm';
 import rehypeSanitize from 'rehype-sanitize';
 import { getRankGrade } from '../../utils/search/utils';
 import { getTypeContent } from '../../containers/ipfs/useGetIpfsContentHook';
-import { getContentByCid } from '../../utils/utils-ipfs';
+import { getContentByCid } from '../../utils/ipfs/utils-ipfs';
 import SearchItem from '../SearchItem/searchItem';
+import useIpfs from 'src/hooks/useIpfs';
+import { $TsFixMe } from 'src/types/tsfix';
 
-function ContentItem({ item, cid, nodeIpfs, grade, ...props }) {
-  const [content, setContent] = useState(null);
-  const [textPreview, setTextPreview] = useState(cid);
-  const [typeContent, setTypeContent] = useState('');
+type ContentItemProps = {
+  item: $TsFixMe;
+  cid: string;
+  grade?: $TsFixMe;
+  className?: string;
+};
+
+function ContentItem({ item, cid, grade, className }: ContentItemProps) {
+  const [content, setContent] = useState<string | undefined>(undefined);
```

Comment:
can't `undefined` be ommited? looks unnessary

Reply:
and type

Reply:
a lot of typescript things - more difficult to read

Reply:
At this stage it necessary, after that will be refactored